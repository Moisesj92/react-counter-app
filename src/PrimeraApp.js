//Componentes funcionales
import React from 'react';
import PropTypes from 'prop-types';
//import React, {Fragment} from 'react';

//Agregar valores por defecto es una bueno practica
//const PrimeraApp = ({ saludo = 'Hola Mundo' }) => {
const PrimeraApp = ({ saludo, subtitulo }) => {


    //Obligar a que vengan las props que necesito


    return (
        <>
            <h1> { saludo } </h1>
            <p> { subtitulo } </p>
        </>
    ); 
    
}

//Obligar a que vengan las props que necesito
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    saludo: 'hola mundo',
    subtitulo: 'soy un subtitulo 1234'
}

export default PrimeraApp;