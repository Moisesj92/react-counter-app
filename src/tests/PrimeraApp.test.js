import React from 'react';
import '@testing-library/jest-dom'
import {shallow} from 'enzyme';
import PrimeraApp from '../PrimeraApp';

describe( 'Prueba en <PrimeraApp />', () => {


    test('Debe mostrar <PrimeraApp /> correctamente', () =>{

        const saludo = 'hey hey';

        const wrapper = shallow( <PrimeraApp saludo={saludo}/> );

        expect( wrapper ).toMatchSnapshot();

    });

});