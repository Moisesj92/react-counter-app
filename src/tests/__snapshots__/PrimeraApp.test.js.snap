// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    hey hey
     
  </h1>
  <p>
     
    soy un subtitulo 1234
     
  </p>
</Fragment>
`;
